satellite/analytics/service.go

```diff
@@ -48,22 +63,34 @@ func NewService(log *zap.Logger, config Config, satelliteName string) *Service {
 		config:        config,
 		satelliteName: satelliteName,
 		clientEvents:  make(map[string]bool),
+		startChan:     make(chan struct{}),
 	}
 	if config.Enabled {
 		service.segment = segment.New(config.SegmentWriteKey)
 	}
 	for _, name := range []string{eventGatewayCredentialsCreated, eventPassphraseCreated, eventExternalLinkClicked, eventPathSelected, eventLinkShared} {
 		service.clientEvents[name] = true
 	}
+
 	return service
 }
 
+// Run runs the service and use the context in new requests.
+func (service *Service) Run(ctx context.Context) error {
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
WithCancel is not necessary here, since `Close` will be closed only after `ctx` has been cancelled.